macOS: linbox-1.7.0 fails to build

### Steps To Reproduce

Possibly specific to x86 / OSX 14, not sure:
```
[fflas_ffpack-2.5.0] [spkg-install] libtool: link: g++ -std=gnu++11 -std=gnu++11 -dynamiclib  -o .libs/libfflas_c.1.dylib  .libs/fflas_lvl1.o .libs/fflas_lvl2.o .libs/fflas_lvl3.o .libs/fflas_sparse.o   -L/Users/kwankyu/GitHub/sage-dev/local/lib ./.libs/libfflas.dylib -lopenblas  -O2 -march=native -g -O2 -Wl,-rpath -Wl,/Users/kwankyu/GitHub/sage-dev/local/lib -Wl,-rpath -Wl,/Users/kwankyu/GitHub/sage-dev/local/lib   -install_name  /Users/kwankyu/GitHub/sage-dev/local/lib/libfflas_c.1.dylib -compatibility_version 2 -current_version 2.0 -Wl,-single_module
[fflas_ffpack-2.5.0] [spkg-install] Undefined symbols for architecture x86_64:
[fflas_ffpack-2.5.0] [spkg-install]   "Givaro::Integer::operator%(unsigned long long) const", referenced from:
[fflas_ffpack-2.5.0] [spkg-install]       Givaro::Modular<double, double, void>::init(double&, Givaro::Integer const&) const in fflas_lvl1.o
[fflas_ffpack-2.5.0] [spkg-install]       Givaro::Modular<double, double, void>::init(double&, Givaro::Integer const&) const in fflas_lvl2.o
[fflas_ffpack-2.5.0] [spkg-install]       Givaro::Modular<double, double, void>::init(double&, Givaro::Integer const&) const in fflas_lvl3.o
[fflas_ffpack-2.5.0] [spkg-install] ld: symbol(s) not found for architecture x86_64
[fflas_ffpack-2.5.0] [spkg-install] clang: error: linker command failed with exit code 1 (use -v to see invocation)
[fflas_ffpack-2.5.0] [spkg-install] make[7]: *** [libfflas_c.la] Error 1
[fflas_ffpack-2.5.0] [spkg-install] make[7]: *** Waiting for unfinished jobs....
[fflas_ffpack-2.5.0] [spkg-install] make[6]: *** [all-recursive] Error 1
[fflas_ffpack-2.5.0] [spkg-install] make[5]: *** [all-recursive] Error 1
[fflas_ffpack-2.5.0] [spkg-install] make[4]: *** [all-recursive] Error 1
[fflas_ffpack-2.5.0] [spkg-install] make[3]: *** [all] Error 2
[fflas_ffpack-2.5.0] [spkg-install] ********************************************************************************
[fflas_ffpack-2.5.0] [spkg-install] Error building fflas_ffpack-2.5.0
```
Adding `-lgivaro` to the linker command line fixes at least that command

### Expected Behavior

Builds

### Actual Behavior

fflas_ffpack-2.5.0 fails to build

### Additional Information

_No response_

### Environment

```markdown
- **OS**: macOS 
- **Sage Version**: 10.4.beta6
```


### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

--- a/build/pkgs/fflas_ffpack/spkg-install.in
+++ b/build/pkgs/fflas_ffpack/spkg-install.in
@@ -29,6 +29,13 @@ if [ -z "$CONFIG_SHELL" ]; then
     export CONFIG_SHELL=`command -v bash`
 fi
 
+# Dirty	workaround to link on OSX
+# https://github.com/linbox-team/fflas-ffpack/issues/391
+# https://github.com/sagemath/sage/issues/38002
+if [ "$UNAME" = Darwin ]; then
+    LDFLAGS="-lgivaro $LDFLAGS"
+fi
+
 # We disable openmp because of build failures, see
 # https://github.com/sagemath/sage/issues/17635#comment:67
 sdh_configure --with-default="$SAGE_LOCAL" --with-blas-libs="$LINBOX_BLAS" \

--- a/build/pkgs/linbox/patches/310-backport.patch
+++ b/build/pkgs/linbox/patches/310-backport.patch
@@ -27,7 +27,7 @@ index 59006d6c5f..2604f47b81 100644
 +				const_cast<data_it>(_data_beg)  = iter._data_beg ;
 +				const_cast<data_it>(_data_end)  = iter._data_end  ;
 +				const_cast<Field &>(_field)     = iter._field ;
-+				const_cast<size_t&>(ld)         = iter._ld ;
++				const_cast<size_t&>(_ld)         = iter._ld ;
  				_row       = iter._row ;
  
  				return *this;
@@ -48,7 +48,7 @@ index 498a5525db..a60943868b 100644
 +				const_cast<element_iterator>(_data_end)= iter._data_end  ;
 +				const_cast<Field &>(_field) = iter._field  ;
 +				const_cast<std::vector<size_t>&>(_rowid) = iter._rowid;
-+				const_cast<size_t&>(ld) = iter._ld ;
++				const_cast<size_t&>(_ld) = iter._ld ;
  				_row = iter._row ;
  
  				return *this;
@@ -63,7 +63,7 @@ index 498a5525db..a60943868b 100644
 +				const_cast<data_it>(_data_beg) = iter._data_beg ;
 +				const_cast<data_it>(_data_end) = iter._data_end  ;
 +				const_cast<Field &>(_field)    = iter._field ;
-+				const_cast<size_t&>(ld)= iter._ld ;
++				const_cast<size_t&>(_ld)= iter._ld ;
  				_row       = iter._row ;
  
  				return *this;

